locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+               +> emit .drain();       |  ^Buffer^  |                         |
+               +> emit .resume();      +------------+                         |
+                                       |  ^Buffer^  |                         |
+                                       +------------+   add chunk to queue    |
+                                       |            <---^---------------------<
+                                       +============+
+```
+
+_Note:_ The `.pipe()` function is typically where backpressure is invoked. In
+an isolate application, both [`Readable`][] and [`Writable`][] streams
+should be present. If you are writing an application meant to accept a
+[`Readable`][] stream, or pipes to a [`Writable`][] stream from another
+component, you may omit this detail.
+
+## Example App
+
+Since [Node.js v0.10][], the [`Streams`][] class has offered the ability to
+modify the behaviour of the [`.read()`][] or [`.write()`][] by using the
+underscore version of these respective functions ([`._read()`][] and
+[`._write()`][]).
+
+There are guidelines documented for [implementing Readable streams][] and
+[implementing Writable streams][]. We will assume you've read these over.
+
+This application will do something very simple: take the data source and
+perform something fun to the data, and then write it to another file.
+
+## Rules to Abide For Writing Custom Streams
+
+Recall that a [`.write()`][] may return true or false dependent on some
+conditions. Luckily for us, when building our own [`Writable`][] stream,
+the [`stream state machine`][] will handle our callbacks and determine when to
+handle backpressure and optimize the flow of data for us.
+
+However, when we want to use a [`Writable`][] directly, we must respect the
+`.write()` return value and pay close attention these conditions:
+
+* If the write queue is busy, [`.write()`][] will return false.
+* If the data chunk is too large, [`.write()`][] will return false (the limit
+is indicated by the variable, [`highWaterMark`][]).
+
+_Note_: In most machines, there is a byte size that is determines when a buffer
+is full (which will vary across different machines). Node.js allows you to set
+your own custom [`highWaterMark`][], but commonly, the default is the optimal
+value for what system is running the application. In instances where you might
+want to raise that value, go for it, but do so with caution!
+
+## Build a Writable Stream
+
+When we code a custom [`._write()`][], the code gets used internally by the
+prototypical [`.write()`][]. This does not override the backpressure mechanism,
+but we must respect the return value whenever we are building custom streams.
+
+```javascript
+const Writable = require('stream').Writable;
+
+class MyWritable extends Writable {
+  constructor(options) {
+    super(options);
+  }
+
+  _write(chunk, encoding, callback) {
+
+    // does something?
+
+  }
+}
+```
```

Comment:
We need an equal section on 'Write a Readable Stream'.

We should mention that [`readable-stream`][https://github.com/nodejs/readable-stream] is the best way to build a stream application across multiple versions of Node.js, and the browser too! This will make your application have a stable base, and an implementation that changes with its own lifecycle compared to node core.